nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala

```diff
@@ -1540,12 +1540,33 @@ abstract class NirCodeGen
           val offset = genExpr(offsetp, ptr)
           offset withOp Op.Elem(ty, ptr.value, Seq(offset.value))
 
-        case (PTR_SUB, Seq(offsetp, tagp)) =>
-          val st     = unwrapTag(tagp)
-          val ty     = genType(st, box = false)
-          val offset = genExpr(offsetp, ptr)
-          val neg    = negateInt(offset.value, offset)
-          neg withOp Op.Elem(ty, ptr.value, Seq(neg.value))
+        case (PTR_SUB, Seq(argp, tagp)) =>
+          argp match {
+            case lit: Literal =>
+              val st     = unwrapTag(tagp)
+              val ty     = genType(st, box = false)
+              val offset = genExpr(argp, ptr)
+              val neg    = negateInt(offset.value, offset)
+              neg withOp Op.Elem(ty, ptr.value, Seq(neg.value))
+
+            case id: Ident =>
+              def show(f: Focus): Focus = {
+                f.insts.foreach(i => println("  |" + i))
+                f
+              }
+              /*
+
+              With tag type
+              val st     = unwrapTag(tagp)
+              val ty     = genType(st, box = false)
+              val ptrInt    = ptrArg withOp Op.Conv(nir.Conv.Ptrtoint, ty, ptr.value)
+              val ptrArgInt = ptrInt withOp Op.Conv(nir.Conv.Ptrtoint, ty, ptrArg.value)
+              */
+              val ptrArg    = genExpr(argp, ptr)
+              val ptrInt    = ptrArg withOp Op.Conv(nir.Conv.Ptrtoint, nir.Type.ULong, ptr.value)
+              val ptrArgInt = ptrInt withOp Op.Conv(nir.Conv.Ptrtoint, nir.Type.ULong, ptrArg.value)
```

Comment:
@densh This should pass in the pointer type using the tag. Not sure how to do that.

Right now for `int` this generates

```
  %src.57 = ptrtoint i8* %src.43 to i64
  %src.58 = ptrtoint i8* zeroinitializer to i64
```
And it should generate
```
 %46 = ptrtoint i32* %44 to i64
  %47 = ptrtoint i32* %45 to i64
```